iac-extensions/docker/src/main/java/org/sonar/iac/docker/parser/grammar/DockerGrammar.java

```diff
@@ -237,9 +231,33 @@ public LabelInstruction LABEL() {
 
   public EnvInstruction ENV() {
     return b.<EnvInstruction>nonterminal(DockerLexicalGrammar.ENV).is(
-      f.env(b.token(DockerKeyword.ENV),
-        b.oneOrMore(
-          b.firstOf(KEY_VALUE_PAIR_WITH_EQUALS(), KEY_VALUE_PAIR())
+      b.firstOf(
+        f.env(
+          b.token(DockerKeyword.ENV),
+          b.oneOrMore(
+            f.ignoreFirst(
+              b.token(DockerLexicalGrammar.WHITESPACE),
+              KEY_VALUE_PAIR_WITH_EQUAL()
+            )
+          )
+        ),
+        f.env(
+          b.token(DockerKeyword.ENV),
+          b.oneOrMore(
+            f.ignoreFirst(
+              b.token(DockerLexicalGrammar.WHITESPACE),
+              KEY_VALUE_PAIR_WITHOUT_EQUAL()
+            )
+          )
+        ),
+        f.env(
+          b.token(DockerKeyword.ENV),
+          b.oneOrMore(
+            f.ignoreFirst(
+              b.token(DockerLexicalGrammar.WHITESPACE),
+              NEW_KEY_VALUE_PAIR_WITHOUT_VALUE()
+            )
+          )
```

Comment:
1. In such cases it would be smart to use `GrammarBuilder.firstOf()` instead
2. You can reuse `NEW_KEY_ONLY()` instead of a new rule `NEW_KEY_VALUE_PAIR_WITHOUT_VALUE()`
   This would allow a syntax like `ENV key1` however this is invalid syntax and we don't have to care about this.
   But it would parse "key1=" as well as "key1= key2=" also correctly.
3. You can not have multiple KEY_VALUE_PAIR_WITHOUT_EQUAL because every following expression belongs to the first KeyValuePair.